meta/store.go

```diff
@@ -1250,7 +1261,11 @@ func (s *Store) CreateShardGroupIfNotExists(database, policy string, timestamp t
 	// Attempt to create database.
 	sgi, err := s.CreateShardGroup(database, policy, timestamp)
 	if err == ErrShardGroupExists {
-		return s.ShardGroupByTimestamp(database, policy, timestamp)
+		sgi, err = s.ShardGroupByTimestamp(database, policy, timestamp)
+	}
+	// Check that we are returning either an error or a valid shard group.
+	if sgi == nil && err == nil {
+		return nil, errors.New("failed to create a new shard group, error unknown.")
 	}
 	return sgi, err
```

Comment:
This function was returning nil,nil as a result of the bug which later caused the panic.